src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/DataSetService.cs

```diff
@@ -170,66 +184,56 @@ private async Task<DataSetViewModel> MapDataSet(DataSet dataSet, CancellationTok
         };
     }
 
-    private async Task<List<Guid>> GetDataSetPreviousReleaseIds(DataSet dataSet, CancellationToken cancellationToken)
-    {
-        var previousReleaseFileIds = dataSet
-            .Versions
-            .Select(version => version.ReleaseFileId)
-            .ToList();
-
-        return await GetReleaseIdsForReleaseFiles(
-            contentDbContext,
-            previousReleaseFileIds,
-            cancellationToken);
-    }
-
-    private static async Task<List<Guid>> GetReleaseIdsForReleaseFiles(
-        ContentDbContext contentDbContext,
-        List<Guid> releaseFileIds,
+    private async Task<Dictionary<Guid, Dictionary<Guid, ReleaseVersion>>> GetPreviousReleasesByVersionByDataSet(
+        IReadOnlyList<DataSet> dataSets, 
         CancellationToken cancellationToken)
     {
-        return await contentDbContext
+        var dataSetsByPreviousReleaseFileId = dataSets
```

Comment:
Feels like we're missing a trick - we seem to do a lot of this same operation (well, maybe not a lot but enough that I notice it).  Wondering if this is an indicator that there's a complexity in our model that we could simplify somehow, or otherwise create some reusable code to skip around this whole ReleaseFile link.

Reply:
(nothing to do with this PR though - just highlighting it to me again)

Reply:
Are we talking about having to navigate through the related `DataSetVersions`, from the `DataSet`, in order to get the `ReleaseFileId`?

Reply:
Yeah - as I said, nothing to be done in this PR and maybe nothing that needs to be done at all - it's just in the back of my mind that this area of our model feels complex.